*** enchant.js-builds-0.7.0/build/enchant.js	2013-06-07 19:46:24.000000000 +0900
--- enchant.js	2014-02-08 08:08:58.476802608 +0900
***************
*** 7,12 ****
--- 7,17 ----
   */
  
  (function(window, undefined){
+ 	NODE = 0
+ 	CALLSTACK = 0
+ 	CALLSTACK_PEAK = 0
+ 	EVENTLISTENER = 0
+ 	ELEMENT = 0
  
  /**
   * ECMA-262 5th edition Functions
***************
*** 773,780 ****
--- 778,787 ----
          var listeners = this._listeners[type];
          if (listeners == null) {
              this._listeners[type] = [listener];
+ 			//DEBUG("enchant.js/EVENTLISTENER: "+(++EVENTLISTENER)+" [+]");
          } else if (listeners.indexOf(listener) === -1) {
              listeners.unshift(listener);
+ 			//DEBUG("enchant.js/EVENTLISTENER: "+(++EVENTLISTENER)+" [+]");
  
          }
      },
***************
*** 798,803 ****
--- 805,811 ----
              var i = listeners.indexOf(listener);
              if (i !== -1) {
                  listeners.splice(i, 1);
+ 				//DEBUG("enchant.js/EVENTLISTENER: "+(--EVENTLISTENER)+" [+]");
              }
          }
      },
***************
*** 808,813 ****
--- 816,822 ----
       */
      clearEventListener: function(type) {
          if (type != null) {
+ 			//DEBUG("enchant.js/EVENTLISTENER: "+(EVENTLISTENER -= this._listeners[type].length)+" [+]");
              delete this._listeners[type];
          } else {
              this._listeners = {};
***************
*** 914,924 ****
                  while (stage.firstChild) {
                      stage.removeChild(stage.firstChild);
                  }
!                 stage.style.position = 'relative';
  
                  var bounding = stage.getBoundingClientRect();
!                 this._pageX = Math.round(window.scrollX || window.pageXOffset + bounding.left);
!                 this._pageY = Math.round(window.scrollY || window.pageYOffset + bounding.top);
              }
              stage.style.fontSize = '12px';
              stage.style.webkitTextSizeAdjust = 'none';
--- 923,933 ----
                  while (stage.firstChild) {
                      stage.removeChild(stage.firstChild);
                  }
!                 //stage.style.position = 'relative';
  
                  var bounding = stage.getBoundingClientRect();
!                 this._pageX = Math.round((window.scrollX || window.pageXOffset) + bounding.left);
!                 this._pageY = Math.round((window.scrollY || window.pageYOffset) + bounding.top);
              }
              stage.style.fontSize = '12px';
              stage.style.webkitTextSizeAdjust = 'none';
***************
*** 1493,1498 ****
--- 1502,1510 ----
              this.frame++;
              now = window.getTime();
              this.currentTime = now;
+ 
+ 			//this.frame += parseInt(1000 / this.fps / (now - this._calledTime))
+ 			//this._requestNextFrame(1000 / this.fps % (now - this._calledTime));
              this._requestNextFrame(1000 / this.fps - (now - this._calledTime));
          },
          getTime: function() {
***************
*** 2730,2735 ****
--- 2742,2748 ----
                          }
                          increase = increase / 2 | 0;
                      }
+                     //ctx.strokeText(buf, 0, y);
                      ctx.fillText(buf, 0, y);
                      y += line.height - 1;
                      c += length;
***************
*** 2742,2747 ****
--- 2755,2761 ----
                  } else {
                      x = 0;
                  }
+                 //ctx.strokeText(buf, x, y);
                  ctx.fillText(buf, x, y);
                  y += line.height - 1;
              }
***************
*** 3223,3228 ****
--- 3237,3243 ----
              var addedToScene = new enchant.Event('addedtoscene');
              node.dispatchEvent(addedToScene);
          }
+ 		DEBUG("enchant.js/NODE: "+(++NODE)+" [+]");
      },
      /**
       * Incorporates Node into Group.
***************
*** 3269,3274 ****
--- 3284,3297 ----
                  var removedFromScene = new enchant.Event('removedfromscene');
                  node.dispatchEvent(removedFromScene);
              }
+             if (this._element)
+                 node._element = null;
+             if (this.context)
+                 node.context = null;
+ 			while(node.childNodes != null && node.childNodes.length > 0)
+ 				node.removeChild(node.childNodes[0])
+ 			node.clearEventListener()
+ 			DEBUG("enchant.js/NODE: "+(--NODE)+" [-]");
          }
      },
      /**
***************
*** 4624,4630 ****
          this.context = null;
  
          var id = 'enchant-surface' + core._surfaceID++;
!         if (document.getCSSCanvasContext) {
              this.context = document.getCSSCanvasContext('2d', id, width, height);
              this._element = this.context.canvas;
              this._css = '-webkit-canvas(' + id + ')';
--- 4647,4653 ----
          this.context = null;
  
          var id = 'enchant-surface' + core._surfaceID++;
!         if (0 && document.getCSSCanvasContext) {
              this.context = document.getCSSCanvasContext('2d', id, width, height);
              this._element = this.context.canvas;
              this._css = '-webkit-canvas(' + id + ')';
***************
*** 5806,5816 ****
--- 5829,5849 ----
              e.timeline = this;
              action.dispatchEvent(e);
          }
+ 		//++CALLSTACK;
          if (remainingTime > 0 || (this.queue[0] && this.queue[0].time === 0)) {
+         //if (++CALLSTACK < 2048 && (remainingTime > 0 || (this.queue[0] && this.queue[0].time === 0))) {
+ 			CALLSTACK_PEAK = CALLSTACK
+ 			//DEBUG("enchant.js/CALLSTACK: "+(CALLSTACK)+" [+]");
+ 			//if(CALLSTACK >= 256)
+ 			//	DEBUG("enchant.js/CALLSTACK: Peak "+(CALLSTACK));
              var event = new enchant.Event("enterframe");
              event.elapsed = remainingTime;
              this.dispatchEvent(event);
+ 			if(CALLSTACK_PEAK === CALLSTACK && CALLSTACK_PEAK >= 256)
+ 				DEBUG("enchant.js/CALLSTACK: Peak "+(CALLSTACK_PEAK));
          }
+ 		//DEBUG("enchant.js/CALLSTACK: "+(--CALLSTACK)+" [-]");
+ 		--CALLSTACK;
      },
      /**
       */
